engine/src/filter.rs

```diff
@@ -13,16 +13,16 @@ pub struct SchemeMismatchError;
 // under the hood propagates field values to its leafs by recursively calling
 // their `execute` methods and aggregating results into a single boolean value
 // as recursion unwinds.
-pub(crate) struct CompiledExpr<'s>(Box<dyn 's + Fn(&ExecutionContext<'s>) -> bool>);
+pub(crate) struct CompiledExpr<'s>(Box<dyn 's + Fn(&ExecutionContext<'s>) -> Option<bool>>);
 
 impl<'s> CompiledExpr<'s> {
     /// Creates a compiled expression IR from a generic closure.
-    pub(crate) fn new(closure: impl 's + Fn(&ExecutionContext<'s>) -> bool) -> Self {
+    pub(crate) fn new(closure: impl 's + Fn(&ExecutionContext<'s>) -> Option<bool>) -> Self {
         CompiledExpr(Box::new(closure))
     }
 
     /// Executes a filter against a provided context with values.
-    pub fn execute(&self, ctx: &ExecutionContext<'s>) -> bool {
+    pub fn execute(&self, ctx: &ExecutionContext<'s>) -> Option<bool> {
```

Comment:
This should always return `bool` instead of propagating upwards. Direct comparisons should return `false` in case of a missing LHS.

Reply:
Except for the `NotEqual` right? `<nil> != "wirefilter"` is `true` ?

Reply:
I'm not sure. Check the wireshark behaviour for consistency.

Reply:
So the funny thing is that Wireshark itself is inconsistent:
![Wireshark_inequal](https://user-images.githubusercontent.com/1629419/57076429-8e28f100-6ce1-11e9-94ef-4c170371fd32.png)
![Wireshark_not_equal](https://user-images.githubusercontent.com/1629419/57076430-8e28f100-6ce1-11e9-93b5-d550f281ab7a.png)
It sounds like `<nil> != <value>` is also false.

Reply:
Code was updated as long as the PR description.